internal/engine/eval/rego/datasources.go

```diff
@@ -56,6 +57,7 @@ func buildFromDataSource(
 			}
 
 			if err := dsf.ValidateArgs(jsonObj); err != nil {
+				log.Error().Interface("dsfunction", k).Err(err).Msg("error validating datasource function arguments")
```

Comment:
**nit:** I would avoid logging so deep in the stack, it would be much better to raise a proper error and log in upper layers instead.

Given that Data Sources are meant to be user-defined, I don't think there's much value to log an error, as the user would not have access to it. Besides, we have no idea how often this function is going to be called by REGO code in the rule type, this might be spammy.

Reply:
This is mostly for us to track the rego eval engine, it's hard to observe what it's doing without them. Would it be better if I set the level to Debug?